Tests/AppTests/RegressionTests.swift

```diff
@@ -0,0 +1,29 @@
+@testable import App
+
+import ShellOut
+import XCTVapor
+
+
+class RegressionTests: XCTestCase {
+
+    func test_issue_58_git_prompt_crash() throws {
+        // https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/issues/58
+        try XCTSkipIf(isRunningInCI)  // doesn't work in CI
+
+        Current.shell = .live
+        Current.fileManager = .live
+        // specifying a non-existant repo will trigger a 404, which will trigger a prompt
+        // unfortunately, this does *not* reproduce the hanging when prompts are allowed,
+        // presumably because the test runs without interactive terminal
+        let pkg = Package(url: "https://github.com/foo-nonexistant/bar.git")
+        let cacheDir = try XCTUnwrap(Current.fileManager.cacheDirectoryPath(for: pkg))
+        let wdir = Current.fileManager.checkoutsDirectory
+        XCTAssertThrowsError(
+            try Current.shell.run(command: .gitClone(url: URL(string: pkg.url)!, to: cacheDir), at: wdir)
+        ) { error in
+            let e = error as? ShellOutError
+            XCTAssertEqual(e?.message, "fatal: could not read Username for 'https://github.com': Device not configured")
```

Comment:
This is key here for the diagnosis: `Device not configured`. This only happens when run via tests or in a container and that's why it crashes if you suppress the prompt in a container, I suppose.

Reply:
BTW, I'm not sure if the test adds any value as is - it's not really testing our error case. Should I take it out?

Reply:
Removed as discussed!